crates/astria-composer/src/searcher/api.rs

```diff
@@ -0,0 +1,58 @@
+use std::net::SocketAddr;
+
+use axum::{
+    extract::State,
+    response::IntoResponse,
+    routing::get,
+    Router,
+};
+use serde::Serialize;
+use thiserror::Error;
+
+use super::State as SearcherState;
+
+#[derive(Debug, Error)]
+pub enum ApiError {
+    #[error("hyper error")]
+    ServerError(#[from] hyper::Error),
+}
+
+pub(super) async fn run(api_url: &SocketAddr, state: &SearcherState) -> Result<(), ApiError> {
+    let api_router = Router::new()
+        .route("/healthz", get(healthz))
+        .with_state(state.clone()); // TODO: should use an Arc here instead of cloning, otherwise state is stale
```

Comment:
You probably want a `Arc<State>` instead of a `State(Arc<T>)`, but it's hard to tell without knowing what enters the state. Try to express what kind of state you want to maintain and what state is supposed to be shared between API calls?